### PR TITLE
Update from System.Data.SqlClient to Microsoft.Data.SqlClient in v1.x

### PR DESCRIPTION
**Pull request by janrybka**
Details in of update in discussion: https://github.com/eventflow/EventFlow/discussions/1022#discussion-6435423

This is first PR. The second will be to `develop-v0`. Unfortunately I don't have any sample project to run tests on. Changes looks straight forward, build and unit tests passes, so there shouldn't be problems with runtime library mismatches. 

I'm currently running test on v0 version. I'll post those once finished.

### CODE REVIEW

**Comment by janrybka** [comment]
I see I need to disable cert validation on SQL connection in tests... Work ongoing...

... Done

File: Source/EventFlow.TestHelpers/MsSql/MsSqlHelpz.cs [hunk 1]
```diff
@@ -50,7 +50,8 @@ public static MsSqlConnectionString CreateConnectionString(string label)
                 {
                     DataSource = FirstNonEmpty(
                         Environment.GetEnvironmentVariable("EVENTFLOW_MSSQL_SERVER"),
-                        ".")
+                        "."),
+                    Encrypt = false
```
Comment (janrybka) [inline Source/EventFlow.TestHelpers/MsSql/MsSqlHelpz.cs:53]: [Trust Server Cert](https://learn.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlconnectionstringbuilder.trustservercertificate?view=dotnet-plat-ext-8.0) wasn't working (I assume with SQL Server 2017), so I just disabled encryption. It's not required for tests.

**Comment by rasmus** [comment]
@janrybka there seems to be something odd with the CLA. Are your commits done using an email verified on GitHub?

**Comment by janrybka** [comment]
> @janrybka there seems to be something odd with the CLA. Are your commits done using an email verified on GitHub?

Yes, I used the wrong one on this repo. Fixed.

**Comment by rasmus** [comment]
❤️

**Pull request merged**